parser.js

```diff
@@ -36,16 +36,24 @@ Parser.prototype._newPacket = function () {
 }
 
 Parser.prototype.parse = function (buf) {
+
   this._list.append(buf)
 
-  while ((this.packet.length != -1 || this._list.length > 0) &&
-         this[this._states[this._stateCounter]]()) {
-    this._stateCounter++
+  try {
+
+    while ((this.packet.length != -1 || this._list.length > 0) &&
+           (noError = this[this._states[this._stateCounter]]())) {
+      this._stateCounter++
 
-    if (this._stateCounter >= this._states.length) {
-      this._stateCounter = 0
+      if (this._stateCounter >= this._states.length) {
+        this._stateCounter = 0
+      }
     }
   }
+  catch (e) {
+
+    return this.emit('error', e);
+  }
```

Comment:
No it's not needed. All potental pitfalls i have spotted are checked for explicitely now.

Let's talk about that at another place, we should really introduce a catch *somewhere*, any bug within the processing should only kill the specific client connection and not the whole broker